src/dmd/denum.d

```diff
@@ -342,6 +342,19 @@ extern (C++) final class EnumDeclaration : ScopeDsymbol
                 Loc locx = loc.isValid() ? loc : this.loc;
                 memtype = memtype.typeSemantic(locx, _scope);
             }
+            else
+            {
+                // Try to get base type from the first member, as an alternative to
+                // running semantic at this point, i.e. dsymbolSemantic(this, _scope)
+                if (!isAnonymous() && members && members.length && (*members)[0].isEnumMember().value)
+                {
+                    auto e = (*members)[0].isEnumMember().value;
+                    e = e.expressionSemantic(_scope);
+                    e = resolveProperties(_scope, e);
+                    e = e.ctfeInterpret();
+                    memtype = e.type;
```

Comment:
Couldn't this call `getDefaultValue` (or a refactored method implementing the primary analysis) instead of partially duplicating it's implementation?

Reply:
`getDefaultValue()` calls `dsymbolSemantic(this, _scope);` we don't need to do that yet.

Reply:
Hence the hint regarding a refactored method. This copies parts of `dsymbolSemantic` for `EnumMember`s while omitting e.g. the type painting, no?

Reply:
What type painting? The only thing I see I could do is to assign back `e` ( `em.value = e`) to prevent evaluating it again in `dsymbolsem.d`.

Reply:
E.g. [this code](https://github.com/dlang/dmd/blob/c750659c5235ba5a045eb5e709f9e70c4bb0e2c0/src/dmd/dsymbolsem.d#L2620-L2657) will now be skipped in case of a forward reference because `memtype` is set here?

Reply:
I think it's better to do the complete semantic pass for now, it's more solid ground. Besides, I've already discovered another bug expanding the test and trying to refactor some code.

Reply:
Thanks for the review.